### PR TITLE
Replace ledger migration tool's view with fvm/utils' SimpleView

### PR DESCRIPTION
**Pull request by pattyshack**
This is needed to unblock #3644 since the original migration view does not fully implement the view api.

gh: #3102

### CODE REVIEW

**Review by janezpodhostnik — APPROVED** [review]

File: fvm/utils/view.go [hunk 1]
```diff
@@ -137,3 +170,19 @@ func (m *MapLedger) Delete(owner, key string) error {
 	delete(m.RegisterTouches, flow.RegisterID{Owner: owner, Key: key})
 	return nil
 }
+
+func (m *MapLedger) Payloads() []ledger.Payload {
+	ret := make([]ledger.Payload, 0, len(m.Registers))
+	for id, val := range m.Registers {
+		keyParts := []ledger.KeyPart{
+			ledger.NewKeyPart(0, []byte(id.Owner)),
+			ledger.NewKeyPart(2, []byte(id.Key)),
```
Comment (zhangchiqing) [inline fvm/utils/view.go:179]: Why the type is `2` here?

File: fvm/utils/view.go [hunk 2]
```diff
@@ -137,3 +170,19 @@ func (m *MapLedger) Delete(owner, key string) error {
...
+		}
+
+		ret = append(
+			ret,
+			*ledger.NewPayload(ledger.NewKey(keyParts), ledger.Value(val)))
```
Comment (zhangchiqing) [inline fvm/utils/view.go:184]: It's better to make a `key := KeyFromRegister(id)`, so that we can hide the implementation details of those key parts.

**Review by zhangchiqing — APPROVED** [review]

**Comment by pattyshack** [comment]
bors merge

**Comment by pattyshack** [comment]
bors retry

**Pull request merged**